### ISSUE TITLE
NodeResolverDelegate should return Task<TNode?>

### ISSUE THREAD

**Issue by benmccallum** [ChilliCream/hotchocolate#2015]
I think with nullable reference types turned on, this:
```c#
namespace HotChocolate.Types.Relay
{
    public delegate Task<TNode> NodeResolverDelegate<TNode, TId>(IResolverContext context, TId id);
}
```

Should probably be:
```c#
namespace HotChocolate.Types.Relay
{
    public delegate Task<TNode?> NodeResolverDelegate<TNode, TId>(...
```

Since the relay schema [spec](https://relay.dev/docs/en/graphql-server-specification#schema) for `query.node` is that it returns `Node` (not `Node!`), we should probably flow nullable all the way up. 

This might required a few more nullability changes to flow it all the way through.

**Comment by michaelstaib** [comment]
Yes, we are sprinkling the nullable all over. It will take more time to get it into all the types.

In this case, the `?` will not work since TNode could be a value type.

So, the [MayBeNullAttribute] might be the better choice to fix this.

**Comment by benmccallum** [comment]
> In this case, the ? will not work since TNode could be a value type.

I realised that today and came back here to say haha. Nice one, cheers.

**Comment by benmccallum** [comment]
Fixed by #4615 by @TylerSiegrist, thanks mate! 

(Further to comments above, I doubt a Node can be a scalar, as it'd need to have an id property and thus be an object type).

**Comment by benmccallum** [comment]
The only exception would be if `struct` or `record struct` was supported as a backing type for an object type, but it doesn't seem to be, so I think `TNode` could even be restricted to `class`, but anyway. Good enough :P
